lib/dialects/abstract/query-generator.js

```diff
@@ -914,6 +916,14 @@ class QueryGenerator {
     throw new Error(`Unknown structure passed to order / group: ${util.inspect(collection)}`);
   }
 
+  _updateQuoteIdentifier() {
```

Comment:
I'm trying to wrap my head around this... Do we really need this, or should we just rewrite the original quoteIdentifier to _quoteIdentifier in the class and move this logic to quoteIdentifier itself.

Reply:
So something like this:

Abstract
```
quoteIdentifier(identifier, force) {
  if (identifier === '*') return identifier;
  return this._quoteIdentifier(identifier, force);
}
_quoteIdentifier(identifier, force) {
  throw new Error(`quoteIdentifier for Dialect "${this.dialect}" is not implemented`);
} 
```

mysql
```
  _quoteIdentifier(identifier, force) {
     return Utils.addTicks(Utils.removeTicks(identifier, '`'), '`');
   }
```

Reply:
I have different opinion here:
- `quoteIdentifier` is an external interface 'customer/user' needs to overwrite.
-  The best approach is we use `_quoteIdentifier` in core but that will be an obvious refactor - that is why I come with this mixin like solution.